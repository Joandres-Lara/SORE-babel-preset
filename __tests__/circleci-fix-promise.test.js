const babel = require('@babel/core');
const circleCIFixPromiseModule = require('../src/plugins/circleci-fix-promise-module');

const TEST_TRANSFORM_CODE_WITH_DECLARATION = `
const Promise = require('Promise');
const otherVar = false;

module.exports = function promised(){
 return new Promise((resolve) => {
  return resolve(true);
 });
}
`;

const TEST_TRANSFORM_CODE_WITHOUT_DECLARATION = `
module.exports = function fnWithoutPromiseDeclaration(){
 if(promisedWithNew){
  return new Promise((resolve, reject) => {
   return resolve(true);
  });
 }

 if(promisedWithStaticCalled){
  return Promise.resolve(true);
 }
}
`;

const TEST_TRANSFORM_CODE_WITHOUT_NEED = `
module.exports = function fnNotNeed(){
 return false;
}
`;

describe('Insert promise declaration', () => {

 it('Replace declaration', () => {
  const { code: codeWithDeclaration } = babel.transform(TEST_TRANSFORM_CODE_WITH_DECLARATION, { plugins: [circleCIFixPromiseModule] })
  expect(codeWithDeclaration).toMatchSnapshot();
 });

 it('Declare when is need', () => {
  const { code: codeWithoutDeclaration } = babel.transform(TEST_TRANSFORM_CODE_WITHOUT_DECLARATION, { plugins: [circleCIFixPromiseModule] });
  expect(codeWithoutDeclaration).toMatchSnapshot();
 })

 it('Ignore if not need', () => {
  const { code: codeNotNeedDeclaration } = babel.transform(TEST_TRANSFORM_CODE_WITHOUT_NEED, { plugins: [circleCIFixPromiseModule] });
  expect(codeNotNeedDeclaration).toMatchSnapshot();
 })
})
