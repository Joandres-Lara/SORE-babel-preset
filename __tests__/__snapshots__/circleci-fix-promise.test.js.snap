// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insert promise declaration Declare when is need 1`] = `
"let Promise;

try {
  Promise = require(\\"Promise\\");
} catch (e) {
  console.warn(\\"Module native promise not found, try obtain of babel polyfill configuration\\");
  Promise = require(\\"core-js-pure/features/promise\\");
}

module.exports = function fnWithoutPromiseDeclaration() {
  if (promisedWithNew) {
    return new Promise((resolve, reject) => {
      return resolve(true);
    });
  }

  if (promisedWithStaticCalled) {
    return Promise.resolve(true);
  }
};"
`;

exports[`Insert promise declaration Ignore if not need 1`] = `
"module.exports = function fnNotNeed() {
  return false;
};"
`;

exports[`Insert promise declaration Replace declaration 1`] = `
"let Promise;

try {
  Promise = require(\\"Promise\\");
} catch (e) {
  console.warn(\\"Module native promise not found, try obtain of babel polyfill configuration\\");
  Promise = require(\\"core-js-pure/features/promise\\");
}

const otherVar = false;

module.exports = function promised() {
  return new Promise(resolve => {
    return resolve(true);
  });
};"
`;
